### ISSUE TITLE
Prior Work Discussion

### ISSUE THREAD

**Issue by Necried** [smiths/spreadOverTimeAndTeam#2]
Let me know if we need references on other topics not mentioned here.

TLDR: Many capstone papers, but hard to find those with measures of productivity, and use of team charters. Using GitHub in software classes is relatively new; papers talking about GitHub metrics exist but not for SE capstone. Same story with issue templates.

## SE Capstone papers
There's a lot more generic capstone papers; I just have a few listed here.

### A systematic literature review of capstone courses in software engineering  (2023)
Survey paper of SE capstone courses

### Software Engineering Problems Encountered by Capstone Project Teams (2014)
TLDR given: "Suggestions for mitigating the educationally detrimental situations in capstone projects include an iterative development process, team formation practicalities, reasonable project topics, customereducation, instructions on selecting and adopting crucial tools, emphasizing learning, and adding control to ensure the use of the desired working practices."

### A Reusable, Academic-Strength, Metrics-Based Software Engineering Process for Capstone Courses and Projects (2004)
Describes a "Software Engineering Process" tested in a capstone course

### Academic education of software engineering practices: towards planning and improving capstone courses based upon intensive coaching and team routines (2013)
- Integration of an agile method in a SE capstone course
- Process improvement techniques used such as "coaching" (Which seems to consist of focused standup meetings)

### What Can Students Get from a Software Engineering Capstone Course? (2017)
- A survey conducts on students' initial and final perception on different dimensions involved in their projects
  (technical challenge, teamwork, planning, client negotiation)
- Also follows agile practices
- Statistically significant evidence on the perceived relevance of soft skills changes in students throughout the course

## Team Charters
I couldn't find a specific reference on using team charters in SE courses specifically.

### Laying the foundation for successful team performance trajectories: The roles of team charters and performance strategies (2009)
Study on influences of team charters and performance strategies on performance trajectories of 32 MBA students
competing in a business strategy simulation over time

### Do Team Charters Help Team-Based Projects? The Effects of Team Charters on Performance and Satisfaction in Global Virtual Teams (2021)
From abstract: "These findings suggest team charters may initially aid process improvement but not necessarily the quality of output."

### An empirical study: Team charters and viability in freshmen engineering design (2013)
Team charters in engineering education can improve team effectiveness and viability, providing a valuable return on investment for students without compromising technical content.

### Measuring Productivity of Software Development Teams (2012)
- Lit review paper on techniques and models to measure productivity (no GitHub metrics involved)
- "Software development team productivity can be measured using SLOC, function points, use case points, object points, and feature points, with factors like team size, response time, task complexity, team climate, and cohesion influencing the results."

### Analyzing the Productivity of GitHub Teams based on Formation Phase Activity (2020)
Analysis of productivity at the team formation phase (6 months after repository creation, can fit within a capstone timeframe)

### Student Teamwork on Programming Projects: What can GitHub logs show us? (2020)
From abstract: "We used the students’ GitHub history to classify the student teams into three groups, collaborative, cooperative, or solo-submit, based on the division of labor. We then calculated different metrics for students’ teamwork including the total number and the average number of commits in different parts of the projects and used these metrics to predict the students’ teamwork style. Our findings show that we can identify the students’ teamwork style automatically from their submission logs. This work helps us to better understand novices’ habits while using version control systems. "

### Using GitHub in large software engineering classes. An exploratory case study (2020)
Interesting note here, under Findings: "Different teams adapt GitHub to their workflow differently. Furthermore, despite the steep learning curve, using GitHub should not affect the quality of students’ submissions. *However, using GitHub metrics as a proxy for evaluating team performance can be risky.*"

### Student experiences using GitHub in software engineering courses: a case study (2016)
- Case study on GitHub used in two SE courses
- Note: students in this study concerned that GitHub is not inherently an educational tool; that seems like an outdated perspective

### Using Git to Manage Capstone Software Projects (2012)
- A study on using Git in a capstone course
- Doesn't use metrics; doesn't use GitHub/GitLab

### An Empirical Analysis of Issue Templates Usage in Large-Scale Projects on GitHub (2024)
"Projects with a template exhibited markedly reduced resolution time (381.02 days to 103.18 days) and reduced issue comment count (4.95 to 4.32) compared to those without"

### To Follow or Not to Follow: Understanding Issue/Pull-Request Templates on GitHub (2023)
Investigates the use of Issue/PR templates in open source projects

### Consistent or not? An investigation of using Pull Request Template in GitHub (2022)
From findings: "In this work, we find that only 1.2% of repositories contain the PRT in GitHub, and such repositories are mostly in high popularity and contain a large number of PRs.""

### Learning by doing: Goals and experiences of two software engineering project courses (1994)
Describes two lab class series intended to teach students "reflection in action"

**Comment by Necried** [comment]
I can also commit the bib file of the above if that's OK.

**Comment by smiths** [comment]
@Necried, yes please commit the bib file.  It will be easier to find the papers with the full citation information and not just the title.

This summary is great.  You have shown that there is definitely relevant literature to cite, but you also make a case that what we are doing is sufficiently novel.

**Comment by CSchank** [comment]
I agree! I'm pleasantly surprised that a Git-based, data-driven approach is fairly novel. Thanks so much, Lucas! This paper is increasingly fitting in nicely with my PhD work!

**Comment by Necried** [comment]
Great! The bib file is now up 7fdf175. We can keep this issue open if we need to add more references as the paper gets written, or if I find anything else interesting.

**Comment by smiths** [comment]
This issue was completed so it can be closed.
